storage/mysql/tree_storage.go

```diff
@@ -404,6 +404,13 @@ func (t *treeTX) Rollback() error {
 	return nil
 }
 
+func (t *treeTX) Close() error {
+	if !t.closed {
+		return t.Rollback()
```

Comment:
Errors on rollback are unlikely with an RDBMS but if they happen then it probably means the connection is in a bad state somehow. Either due to a bug or the server going away. This is probably not the right level to detect the latter.

Returning the error looks like the right thing. I generally think all errors should be handled or logged, could be a low severity level and done via deferring a logging helper function. 

I think the errors are probably uninteresting to clients but I'm wary of masking bugs in our code.